To: vim_dev@googlegroups.com
Subject: Patch 7.3.337
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.337 (after 7.3.295)
Problem:    Screen doesn't update after resizing the xterm until a character
	    is typed.
Solution:   When the select call is interrupted check do_resize. (Taylor
	    Hedberg)
Files:	    src/os_unix.c


*** ../vim-7.3.336/src/os_unix.c	2011-09-08 23:24:09.000000000 +0200
--- src/os_unix.c	2011-10-12 20:57:15.000000000 +0200
***************
*** 5146,5156 ****
--- 5146,5163 ----
  # endif
  # ifdef EINTR
  	if (ret == -1 && errno == EINTR)
+ 	{
+ 	    /* Check whether window has been resized, EINTR may be caused by
+ 	     * SIGWINCH. */
+ 	    if (do_resize)
+ 		handle_resize();
+ 
  	    /* Interrupted by a signal, need to try again.  We ignore msec
  	     * here, because we do want to check even after a timeout if
  	     * characters are available.  Needed for reading output of an
  	     * external command after the process has finished. */
  	    goto select_eintr;
+ 	}
  # endif
  # ifdef __TANDEM
  	if (ret == -1 && errno == ENOTSUP)
*** ../vim-7.3.336/src/version.c	2011-10-12 19:53:31.000000000 +0200
--- src/version.c	2011-10-12 21:03:47.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     337,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
64. The remote to the T.V. is missing...and you don't even care.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
